Sources/NIOCore/NIOScheduledCallback.swift

```diff
@@ -170,3 +170,23 @@ extension EventLoop {
         }
     }
 }
+
+@usableFromInline
+struct UnsafeUncheckedScheduledCallbackHandlerWrapper<Handler: NIOScheduledCallbackHandler>: @unchecked Sendable, NIOScheduledCallbackHandler {
```

Comment:
Instead of using `@unchecked Sendable` we should use a NIOLoopBound here instead.

Reply:
Done! [e1e581f](https://github.com/apple/swift-nio/pull/3263/commits/e1e581f1e62b358e06c443d3661ee263e50a0eb7). I wonder if that's the case for `scheduleTask` too where `UnsafeTransfer` (which is `@unchecked Sendable`) is used?